### ISSUE TITLE
Allow us to provide custom Hugo shortcodes, listed in config.json

### ISSUE THREAD

**Issue by xtreme-steve-elliott** [Pivotal-Field-Engineering/pace-builder#18]
One things that has come up for me at least with https://github.com/Pivotal-Field-Engineering/pace-workshop-content/tree/master/dotnet-core-tdd-concourse-cf is that there are going to be multiple flows through the guide, with slight alterations to certain pages. Rather than making multiple versions of the same page, it would be great to have a way of reusing content.

I've spent most of the day doing investigation on the integration with Hugo, and conditionals are possible, albeit only when using HTML page or when you provide custom [Shortcodes](https://gohugo.io/content-management/shortcodes/). I've tested that locally, and having the ability to use shortcodes with the Markdown generated from Pace would both work and likely be the least intrusive of options to allow conditionals.

In premise, what would need to happen would be to add a section to `config.json` like:
```json
"shortcodes": [
  {
    "filename": "<path to shortcode here>/<shortcode name here>.html"
  }
]
```
and when reading it, copy those files into `workshopGen/layouts/shortcodes` after the rest of the `workshopGen` files have been created.

Here's a sample custom shortcode, `contains`, I made for testing:
```html
{{- $name := (.Get 0) -}}
{{- $substring := (.Get 1) -}}
{{- if len .Inner -}}
	{{- $content := .Inner -}}
	{{- with $name -}}
		{{- with ($.Page.Param .) -}}
			{{- if in . $substring -}}
				{{ $content }}
			{{- end -}}
		{{- else -}}
			{{- errorf "Param %q not found: %s" $name $.Position -}}
		{{- end -}}
	{{- else -}}
		{{- errorf "Missing param key: %s" $.Position -}}
	{{- end -}}
{{- end -}}
```

It would be used as:
```markdown
{{< contains title "Concourse" >}}
Concourse specific content here.
{{< /contains >}}
```

Basically it takes the page variable name and the search string, and only renders the inner content if the search string is a substring. It would be incredibly valuable to have at least some access to the broader functionality of Hugo for the workshops built with this.
